designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala

```diff
@@ -2,54 +2,99 @@ package pl.touk.nussknacker.engine.api.deployment.simple
 
 import pl.touk.nussknacker.engine.api.deployment.ProcessActionType.ProcessActionType
 import pl.touk.nussknacker.engine.api.deployment.StateStatus.StatusName
-import pl.touk.nussknacker.engine.api.deployment.{AllowDeployStateStatus, DuringDeployStateStatus, FailedStateStatus, FinishedStateStatus, NotEstablishedStateStatus, ProcessActionType, RunningStateStatus, StateDefinitionDetails, StateStatus}
+import pl.touk.nussknacker.engine.api.deployment.{AllowDeployStateStatus, CustomStateStatus, DuringDeployStateStatus, FinishedStateStatus, NotEstablishedStateStatus, ProcessActionType, RunningStateStatus, StateDefinitionDetails, StateStatus}
+import pl.touk.nussknacker.engine.api.process.VersionId
 
 import java.net.URI
 
 object SimpleStateStatus {
-  val FailedToGet: StateStatus = NotEstablishedStateStatus("FAILED_TO_GET")
-  val Unknown: StateStatus = NotEstablishedStateStatus("UNKNOWN")
+
+  // Represents general problem.
+  case class ProblemStateStatus(description: String, isRedeployable: Boolean = true) extends CustomStateStatus(ProblemStateStatus.name) {
+    override def isFailed: Boolean = true
+  }
+  case object ProblemStateStatus {
+    val name: String = "PROBLEM"
+    val icon: URI = URI.create("/assets/states/error.svg")
+    val defaultDescription = "There are some problems with scenario."
+
+    // Problem factory methods
+
+    def failed: ProblemStateStatus = ProblemStateStatus(defaultDescription)
+
+    def failedToGet: ProblemStateStatus =
+      ProblemStateStatus(errorFailToGetStatus)
+
+    def shouldBeRunning(deployedVersionId: VersionId, user: String): ProblemStateStatus =
+      ProblemStateStatus(shouldBeRunningTooltip(deployedVersionId.value, user))
+
+    def mismatchDeployedVersion(deployedVersionId: VersionId, exceptedVersionId: VersionId, user: String): ProblemStateStatus =
+      ProblemStateStatus(mismatchDeployedVersionTooltip(deployedVersionId.value, exceptedVersionId.value, user))
+
+    def shouldNotBeRunning(deployed: Boolean): ProblemStateStatus =
+      ProblemStateStatus(shouldNotBeRunningMessage(deployed))
+
+    def missingDeployedVersion(exceptedVersionId: VersionId, user: String): ProblemStateStatus =
+      ProblemStateStatus(missingDeployedVersionTooltip(exceptedVersionId.value, user))
+
+    def processWithoutAction: ProblemStateStatus =
+      ProblemStateStatus(processWithoutActionMessage)
+
+    def multipleJobsRunning: ProblemStateStatus =
+      ProblemStateStatus(errorMultipleJobsMessage, isRedeployable = false)
+
+    // Problem descriptions
+
+    private def errorFailToGetStatus: String =
```

Comment:
These belove methods' strings can be moved to the above methods?

Reply:
Fixed.